platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/modelreader/impl/TypeUtils.java

```diff
@@ -36,4 +44,29 @@ public static String extractMethodDescriptor(ExecutableElement methodElement) {
             methodElement.getParameters().stream().map(it -> extractType(it.asType())).toArray(Type[]::new)
         );
     }
+
+    @Nonnull
+    public static List<ExecutableElement> getExecutableElementsFromElements(Stream<? extends Element> elements) {
```

Comment:
💭 These two methods here are just moved from `AbstractInstrumentationProcessor`, so we can reuse them when we read methods on `Adapter`